### ISSUE TITLE
cannot load wallpaper properly in hyprland

### ISSUE THREAD

**Issue by lenglingxuan** [Horus645/swww#100]
To use this as my wallpaper engine, I add in my hyprland.conf as the following.
![image](https://user-images.githubusercontent.com/112380251/223326212-6c52e029-84dd-4cdf-a620-5d6caa71f1e6.png)

But the wallpaper cannot load properly at startup.

the initializtion seems to be successful, because when I run swww init, there appears the following message.
![image](https://user-images.githubusercontent.com/112380251/223326518-ec3c246c-7297-4a81-a5ea-ea2dedb47b86.png)

And when I run the swww img xxx manually, using the same command in my first image, the wallpaper can also load properly.

So is this a bug, or there's something wrong with my exec-once?

**Comment by lenglingxuan** [comment]
> it might be because `swww img` is running too soon, so when it runs the daemon is still starting to fix this u cud put both commands in one exec like so `swww init && swww img ~/wallpapers/1.jpeg` so `swww img` runs only after `swww init` is done

I initially wrote in my config like what you say, but I thought the same way as you, so I thought my config now can solve the problem, which didn't work out. I even added "sleep 2" between the two, but still failed. Just now I changed back as you say, but still failed. But I noticed that, using my exec-once seperately will cause the screen to be completely black, while now there is hyprland logo again, I didn't notice that before.

**Comment by lenglingxuan** [comment]
> it might be because `swww img` is running too soon, so when it runs the daemon is still starting to fix this u cud put both commands in one exec like so `swww init && swww img ~/wallpapers/1.jpeg` so `swww img` runs only after `swww init` is done

Really thanks. I found the problem. It is indeed due to swww img runs too fast that swww init didn't even execute successfully. I once tried adding sleep command in between, and someone once reminded me to connect them using && rather than &, but since I did this in my other parts of exec-once, which don't have any problems, I didn't really pay attention to that. Changing & to && works really well.
